### ISSUE TITLE
Changing description of google_compute_instance_group should not force a recreation

### ISSUE THREAD

**Issue by hype-klaus** [hashicorp/terraform-provider-google#11098]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
terraform --version
Terraform v1.1.4
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.5.0
+ provider registry.terraform.io/hashicorp/google-beta v4.5.0
```

### Affected Resource(s)

* google_compute_instance_group

### Terraform Configuration Files

```tf
provider "google" {
  project = "project"
  region = "region"
  zone = "zone"
}
resource "google_compute_instance_group" "test" {
  name        = "terraform-test"
  description = "Terraform test instance group"
}

# terraform apply and then change description
resource "google_compute_instance_group" "test" {
  name        = "terraform-test"
  description = "Terraform test instance group change"
}
```

### Output

```
# google_compute_instance_group.test must be replaced
-/+ resource "google_compute_instance_group" "test" {
      ~ description = "Terraform test instance group" -> "Terraform test instance group change" # forces replacement
```

### Expected Behavior

Description should get update in-place.

### Actual Behavior

Description change forces a replacement of the resource.

### Steps to Reproduce

1. `terraform apply`
2. change description of google_compute_instance_group
3. `terraform plan`

### References

* #10323

**Comment by rileykarson** [comment]
Note: sizing assumes this is possible, the API may not allow it

**Comment by upodroid** [comment]
@rileykarson  We can close this issue. The API doesn't have a patch/update method to update the instance template.

https://cloud.google.com/compute/docs/reference/rest/beta/instanceTemplates
https://cloud.google.com/compute/docs/reference/rest/v1/instanceTemplates

**Comment by rileykarson** [comment]
You're right that this definitely is not possible: https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups does not have a general update method, just custom ones (& no `setDescription`)
